### PR TITLE
Generate swagger-ui output for client-server API

### PR DESCRIPTION
**Pull request by illicitonion**
Depends on:
https://github.com/matrix-org/matrix-doc/pull/212
https://github.com/matrix-org/matrix-doc/pull/208
https://github.com/matrix-org/matrix-doc/pull/207
for the actual rendered output to not throw javascript errors at runtime.

### CODE REVIEW

File: scripts/add-matrix-org-stylings.sh [hunk 1]
```diff
@@ -16,13 +16,13 @@ for f in "$1"/{head,nav,footer}.html; do
   fi
 done
 
-files=gen/*.html
+files=/home/wadaniel/src/matrix.org/docs/api/client-server/index.html
```
Comment (richvdh) [inline scripts/add-matrix-org-stylings.sh]: no.

Comment (illicitonion) [inline scripts/add-matrix-org-stylings.sh]: Oops!

Done

File: scripts/dump-swagger.py [hunk 2]
```diff
@@ -0,0 +1,61 @@
+#!/usr/bin/env python2
```
Comment (richvdh) [inline scripts/dump-swagger.py]: could you add a comment to say what this does?

File: scripts/dump-swagger.py [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+release_label = sys.argv[2] if len(sys.argv) > 2 else "unstable"
+
+major_version = release_label
+match = re.match("^(r\d)+(\.\d+)?$", major_version)
+if match:
+    major_version = match.group(1)
+
+
+sys.path.insert(0, os.path.join(os.path.dirname(os.path.dirname(os.path.abspath(__file__))), "templating"))
+from matrix_templates.units import resolve_references, MatrixUnits
```
Comment (richvdh) [inline scripts/dump-swagger.py]: generally I prefer to see all imports (and hence any required sys.path wrangling) right at the top of a script. not too  bothered though

File: api/client-server/sync.yaml [hunk 4]
```diff
@@ -310,3 +310,5 @@ paths:
                   "leave": {}
                 }
               }
+      tags:
+        - Room messaging
```
Comment (richvdh) [inline api/client-server/sync.yaml]: This should be in 'room participation' too.

In general the split between 'Room messaging' and 'Room participation' feels fuzzy and arbitrary.

perhaps s/Room participation/Event sync/ ?

File: scripts/dump-swagger.py (hunk 2 above)
Comment (illicitonion) [inline scripts/dump-swagger.py]: Done

File: scripts/dump-swagger.py (hunk 3 above)
Comment (illicitonion) [inline scripts/dump-swagger.py]: Done

File: api/client-server/sync.yaml (hunk 4 above)
Comment (illicitonion) [inline api/client-server/sync.yaml]: Done

Just moved them all to room participatoin

**Comment by richvdh** [comment]
lgtm

**Pull request merged**